tasks/blacklight.rake

```diff
@@ -11,10 +11,15 @@ require 'rubocop/rake_task'
 RuboCop::RakeTask.new(:rubocop)
 
 desc "Run test suite"
-task ci: ['blacklight:generate'] do
+task :ci do
+  system "docker-compose up -d solr"
```

Comment:
Adding this means folks can pull down the Blacklight repo and launch `rake ci` and watch it work, ideally. @cdmo assuming you have docker-compose installed. Can you pull this branch and verify this? If not true, happy to iterate.

Reply:
I ran into an issue where I pulled this branch and all of the feature tests failed for me. `bundle exec rake ci`

Reply:
Also, I noticed that this is both in the `ci` task and the travis file. Is this needed in both placed?

Reply:
I had the same experience as @mejackreed, feature specs failed:

`Finished in 24.6 seconds (files took 4.51 seconds to load)
1140 examples, 71 failures, 6 pending`

Looks like it's unrelated to this PR though if you ask me. I've seen this error before:

```
  64) Facets retains the filters when we change per page number
      Failure/Error: //= link_tree ../images

      ActionView::Template::Error:
        link_tree argument must be a directory
      # ./.internal_test_app/app/assets/config/manifest.js:1
      # ------------------
      # --- Caused by: ---
      # Sprockets::ArgumentError:
      #   link_tree argument must be a directory
      #   ./.internal_test_app/app/assets/config/manifest.js:1
```

I think this is a fix coming #2266 (pretty sure this is caused by sprockets 4.0.0 requiring a change https://github.com/rails/sprockets-rails/issues/443#issuecomment-540696186)

Reply:
@mejackreed is travis just running `bundle exec rake`? I think I was expecting to see a `script` section and didn't, so I guess travis just has that [as a default](https://travis-ci.org/github/projectblacklight/blacklight/jobs/663558443#L470)?

Reply:
I merged in the fix from #2266 and re-ran `ci` and got a new error:

```
  70) Bookmarks navigating from the homepage has a link to the history page
      Failure/Error: <%= javascript_include_tag "application" %>

      ActionView::Template::Error:
        Asset `application.js` was not declared to be precompiled in production.
        Declare links to your assets in `app/assets/config/manifest.js`.

          //= link application.js
        and restart your server
      # ./app/views/layouts/blacklight/base.html.erb:15:in `block in ___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__2593207400865544889_70245572263500'
      # ./app/views/layouts/blacklight/base.html.erb:2:in `___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__2593207400865544889_70245572263500'
      # ./app/views/layouts/blacklight.html.erb:17:in `___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_html_erb__193081276090225314_70245569552620'
      # ./spec/support/features/session_helpers.rb:17:in `sign_in'
      # ./spec/features/bookmarks_spec.rb:6:in `block (3 levels) in <top (required)>'
      # ------------------
      # --- Caused by: ---
      # Sprockets::Rails::Helper::AssetNotPrecompiled:
      #   Asset `application.js` was not declared to be precompiled in production.
      #   Declare links to your assets in `app/assets/config/manifest.js`.
      #
      #     //= link application.js
      #   and restart your server
      #   ./app/views/layouts/blacklight/base.html.erb:15:in `block in ___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__2593207400865544889_70245572263500'
```

Reply:
@mejackreed 
> Also, I noticed that this is both in the `ci` task and the travis file. Is this needed in both places?

I'll try removing the one in `.travis.yml`

Reply:
> @mejackreed is travis just running `bundle exec rake`? I think I was expecting to see a `script` section and didn't, so I guess travis just has that [as a default](https://travis-ci.org/github/projectblacklight/blacklight/jobs/663558443#L470)?

@cdmo yep, that's what is happening, and the default rake task is this `ci` task

Reply:
@cdmo I can get the tests to pass A-OK in this branch. Here is the guide I just wrote up for how I did this:

https://github.com/projectblacklight/blacklight/wiki/Testing-and-Developing-Blacklight